### PR TITLE
build :: set the mvn pmd plugin to fail on violations

### PR DESCRIPTION
**Pull request by dev-mlb**
Run with:
```
mvn clean pmd:check
```
or with the regular build:
```
mvn clean verify
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -897,7 +897,7 @@
           <version>3.16.0</version>
           <configuration>
             <excludes />
-            <failOnViolation>false</failOnViolation>
+            <failOnViolation>true</failOnViolation>
             <failurePriority>1</failurePriority>
```
Comment (dev-mlb) [inline pom.xml:901]: If we fix 3 violations, then we can set this to 2.

**Comment by dev-mlb** [comment]
github build now reports:
```
[INFO] --- maven-pmd-plugin:3.16.0:check (default) @ emissary ---
[INFO] PMD version: 6.42.0
[INFO] You have 692 warnings. For more details see: /home/runner/work/emissary/emissary/target/pmd.xml
```

File: pom.xml [hunk 2]
```diff
@@ -897,14 +897,23 @@
           <version>3.16.0</version>
```
Comment (dev-mlb) [inline pom.xml:897]: Version 3.20.0 is available. I can update it with this pr or wait for a follow-on. I just need to know if it is available.

**Review by cfkoehler — APPROVED** [review]
Report looks good to me

**Review by fbruton — APPROVED** [review]

**Pull request merged**